### DISCUSSION TITLE
Add ability to unselect UX options in the block editor

### DISCUSSION THREAD

**Discussion by X3Technology** [vanjarosoftware/Vanjaro.Platform#1718] [General]
I can select a text element and then pick a color from the Options menu.
![set_color](https://user-images.githubusercontent.com/1585717/131203456-f342f692-f486-43f5-ba54-afe271bb8edf.PNG)

However, what if I want to remove the "text-primary" class from the text element so it defaults to whatever the default text color is?

Right now, it appears the only option to select another preset color (secondary, tertiary, etc.).  There is a workaround to go into the styling tab and delete the 'text-primary" class, but this feels weird.
![remove_text_color](https://user-images.githubusercontent.com/1585717/131203516-b6315fea-4bbd-4823-92e1-90de705f6aea.PNG)

It would more intuitive if I could simply click the color option in the options UX a second time to unselect it.  Is that possible?

**Comment by X3Technology** [comment]
Actually, you should be to 'unselect' several things on the designer UX.  For example, the preset Styles for buttons or text.   Once a style is selected, it can't be unselected.

**Comment by X3Technology** [comment]
Here is a quick list of things I think we should be able to unselect:
Section -
> Alignment (so it defaults to parent alignment)

Grid -
> Alignment 

Heading - 
> Alignment 
> Color
> Styles

Text -
> Alignment
> Color
> Styles

Button -
> Style (probably needs a label change to avoid confusion with 'Styles' below
> Alignment (this one is already unselectable )
> Color
> Styles

Icon -
> Alignment
> Color
> Color (the second is 'background color', probably needs a label change)
> Border (this control has a selectable 'None' which is not the same as no selection)

Link -
> Alignment
> Color

List -
> List Style ('none' is not the same as no selection)
> Alignment
> Color

Image - 
> Alignment

Divider -
> Alignment
> Color

**Comment by meetmandeep** [comment]
> However, what if I want to remove the "text-primary" class from the text element so it defaults to whatever the default text color is?

Default text color is driven by .text-dark. There is no way to change that other than the workaround you mentioned. 

> It would more intuitive if I could simply click the color option in the options UX a second time to unselect it. Is that possible?

No. That's only intuitive for developers or those who understand CSS or web design in general. For a normal end user, it doesn't make any sense to remove all colors and yet still see a color. 

> Actually, you should be to 'unselect' several things on the designer UX. For example, the preset Styles for buttons or text. Once a style is selected, it can't be unselected.

For advanced users, you can always remove the class from styling tab.

**Comment by X3Technology** [comment]
One could argue that the 'normal end user' is also not going to understand things CSS properties like flex, transform, etc. etc. but those are exposed in the UX.

Developers are going to want the ability to have the text inherit the color rather than manually set it every time, or use the workaround of deleting the class from the Styling UX.

Most of the options on the Styling tab already have a tiny 'x' next to the property to clear out the selection if it differs from the default.
![Styling](https://user-images.githubusercontent.com/1585717/131418468-1729d9ea-b909-49f2-b810-f7e8477b1289.PNG)

I think it makes sense to have this 'unselect' functionality on the Options tab as well.

**Comment by meetmandeep** [comment]
> One could argue that the 'normal end user' is also not going to understand things CSS properties like flex, transform, etc. etc. but those are exposed in the UX.

Agreed. Hence, these options are tucked in the Styling Tab for advanced users. You can easily remove a class from Styling Tab.
